packages/flutter/lib/src/gestures/recognizer.dart

```diff
@@ -497,12 +497,16 @@ enum GestureRecognizerState {
 /// in the gesture arena, the gesture will be rejected.
 abstract class PrimaryPointerGestureRecognizer extends OneSequenceGestureRecognizer {
   /// Initializes the [deadline] field during construction of subclasses.
+  /// Both preAcceptSlopTolerance postAcceptSlopTolerance can be:
+  /// - -1 (default) for letting flutter figuring out
+  /// - 0 or a positive number to specify radius in logical pixels
+  /// - null (passed explicitly) not to enforce slop restriction
```

Comment:
i would recommend using null for automatic, and 2^63-1 as "no restriction".

Reply:
that would be a much more clean API, however there's an issue with that approach, and it's that the null value is currently the sentinel for *no restriction*. It will break any existing code that subclasses PrimaryPointerGestureRecognizer using an explicit null for those parameters. Within flutter package, there's LongPressGestureRecognizer that can be adapted to pass double.maxFinite, but outside flutter, on any real project, this will break behavior.

Reply:
@Hixie @goderbauer @jonahwilliams any thoughts about this?

Reply:
@Hixie @goderbauer @jonahwilliams what do you think about this? would you prefer to close this? or keep moving forward?

Reply:
I don't really like the changes to this API as designed, I would suggest either:

1. Keep it as it is today
2. Go through a breaking change process to give it a reasonable API

Reply:
ok, I'll proceed to close this PR, as it's not needed for either of those options